### PR TITLE
refactor: rewrite useBillingAndUsageData with useSubscription and useUserTrial

### PR DESCRIPTION
**Pull request by ridvanakca**

### CODE REVIEW

File: packages/backend/src/graphql/schema.graphql [hunk 1]
```diff
@@ -563,11 +562,6 @@ type License {
   verified: Boolean
 }
 
-type GetBillingAndUsage {
-  subscription: Subscription
-  usage: Usage
-}
-
 type MonthlyQuota {
```
Comment (barinali) [inline packages/backend/src/graphql/schema.graphql:565]: We should remove other types that are only used with `Subscription` as well, like MonthlyQuota, NextBIllAmount etc.

**Review by barinali — CHANGES_REQUESTED** [review]

File: packages/web/src/hooks/useSubscription.ee.js [hunk 2]
```diff
@@ -16,23 +33,38 @@ export default function useSubscription() {
...
+    [checkoutCompleted, hasSubscription],
+  );
+
+  React.useEffect(
+    function stopPollingWhenSubscriptionIsCreated() {
+      if (checkoutCompleted && hasSubscription) {
+        setIsPolling(false);
+      }
+    },
+    [checkoutCompleted, hasSubscription],
```
Comment (kattoczko) [inline packages/web/src/hooks/useSubscription.ee.js:58]: This two useEffects could be limited to just one. For example: 

`if (checkoutCompleted) {
        setIsPolling(!hasSubscription);
  }`

**Review by kattoczko — CHANGES_REQUESTED** [review]
In the Paddle.ee.jsx file the old graphql queries are used:

line: 32
```
await apolloClient.refetchQueries({
      include: ['GetTrialStatus', 'GetBillingAndUsage'],
 });
```

**Review by kattoczko — APPROVED** [review]

File: packages/web/src/components/UsageDataInformation/index.ee.jsx [hunk 3]
```diff
@@ -40,42 +48,94 @@ function BillingCard(props) {
...
+    queryClient.invalidateQueries({
+      queryKey: ['planAndUsage', currentUserId],
+    });
+  }, [subscription, queryClient, currentUserId]);
+
+  if (trial.hasTrial) {
+    billingInfo = {
+      monthlyQuota: {
+        title: formatMessage('usageDataInformation.freeTrial'),
+        action: '/settings/billing/upgrade',
```
Comment (barinali) [inline packages/web/src/components/UsageDataInformation/index.ee.jsx:105]: `URLS.SETTINGS_PLAN_UPGRADE` should be used here.

**Review by barinali — CHANGES_REQUESTED** [review]

**Review by barinali — APPROVED** [review]

**Pull request merged**